rskj-core/src/test/java/co/rsk/pcc/AltBN128Test.java

```diff
@@ -0,0 +1,718 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+import co.rsk.config.TestSystemProperties;
+import co.rsk.config.VmConfig;
+import co.rsk.helpers.Stopwatch;
+import co.rsk.vm.BytecodeCompiler;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.BlockFactory;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.PrecompiledContracts;
+import org.ethereum.vm.VM;
+import org.ethereum.vm.program.Program;
+import org.ethereum.vm.program.Stack;
+import org.ethereum.vm.program.invoke.ProgramInvokeMockImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.util.HashSet;
+
+import static org.ethereum.util.ByteUtil.stripLeadingZeroes;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Sebastian Sicardi
+ * @since 10.09.2019
+ */
+
+public class AltBN128Test {
+    private final TestSystemProperties config = new TestSystemProperties();
+
+    private final PrecompiledContracts precompiledContracts = new PrecompiledContracts(config, null);
+    private ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+    private final VmConfig vmConfig = config.getVmConfig();
+    private final BlockFactory blockFactory = new BlockFactory(config.getActivationConfig());
+    private ProgramInvokeMockImpl invoke;
+    private BytecodeCompiler compiler;
+
+    @Before
+    public void setup() {
+        invoke = new ProgramInvokeMockImpl();
+        compiler = new BytecodeCompiler();
+        when(activations.isActive(ConsensusRule.RSKIP137)).thenReturn(true);
+    }
+
+    @Test
+    public void testCorrectAddressForALTBN128Contracts() {
+        DataWord altBN128AddAddr = PrecompiledContracts.ALT_BN_128_ADD_DW;
+        DataWord altBN128MulAddr = PrecompiledContracts.ALT_BN_128_MUL_DW;
+        DataWord altBN128PairAddr = PrecompiledContracts.ALT_BN_128_PAIRING_DW;
+
+        PrecompiledContracts.PrecompiledContract altBN128Add = precompiledContracts.getContractForAddress(activations, altBN128AddAddr);
+        PrecompiledContracts.PrecompiledContract altBN128Mul = precompiledContracts.getContractForAddress(activations, altBN128MulAddr);
+        PrecompiledContracts.PrecompiledContract altBN128Pair = precompiledContracts.getContractForAddress(activations, altBN128PairAddr);
+
+        Assert.assertNotNull(altBN128Add);
+        Assert.assertNotNull(altBN128Mul);
+        Assert.assertNotNull(altBN128Pair);
+
+        Assert.assertEquals(PrecompiledContracts.BN128Addition.class, altBN128Add.getClass());
```

Comment:
We are trying to use _assertEquals_ which is more JUnit idiomatic

Reply:
You mean `assertThat`?

Reply:
Yes, I'm sorry, that's what I meant